service/src/main/java/bio/terra/pipelines/service/PipelineRunsService.java

```diff
@@ -60,6 +65,78 @@ public PipelineRunsService(
     this.workspaceManagerService = workspaceManagerService;
   }
 
+  /**
+   * Prepare a new PipelineRun for a given pipeline and user-provided inputs. The caller provides a
+   * job uuid and any relevant pipeline inputs. Teaspoons calls WSM to generate a write-only SAS url
+   * for each file input, then persists the pipeline run to the database. Finally, Teaspoons returns
+   * the write SAS url to the user.
+   *
+   * <p>The user can subsequently use the SAS url(s) to upload their input file(s).
+   *
+   * @param pipeline the pipeline to run
+   * @param jobId the job uuid
+   * @param userId the user id
+   * @param userProvidedInputs the user-provided inputs
+   */
+  public Map<String, String> preparePipelineRun(
+      Pipeline pipeline, UUID jobId, String userId, Map<String, Object> userProvidedInputs) {
+
+    PipelinesEnum pipelineName = PipelinesEnum.valueOf(pipeline.getName().toUpperCase());
+
+    if (pipeline.getWorkspaceId() == null) {
+      throw new InternalServerErrorException("%s workspaceId not defined".formatted(pipelineName));
+    }
+
+    // check that we don't already have this jobId
+    if (Boolean.TRUE.equals(pipelineRunsRepository.existsByJobId(jobId))) {
+      throw new DuplicateObjectException(
+          "Duplicate jobId %s found. Please resubmit with a different jobId".formatted(jobId));
+    }
+
+    // define the path for each file input as the jobId + the input key name
+    List<String> userProvidedInputFileKeys =
+        pipeline.getPipelineInputDefinitions().stream()
+            .filter(PipelineInputDefinition::getUserProvided)
+            .filter(p -> p.getType().equals(PipelineInputTypesEnum.VCF))
```

Comment:
at this point we may want to abandon the "VCF" specific type and just use "FILE" since we're not going to be validating suffixes anyway

Reply:
i think we want to filter on `VCF` and `File` either way since we'd want to generate sas tokens for both.  Is there really not a futre where we'll have the ability to validate a users input file names at the level of the code?